File: src/app/model/general-overview.ts
import {Overview} from "./overview";
import {Tags} from "./tags";
import {Images} from "./images";
import {User} from "./user";
import {Rating} from "./rating"
import {LikeAndDislike} from "./likeAndDislike";
import {Comments} from "./comments";

export class GeneralOverview{
  public id!: number;
  public overview: Overview;
  public tags: Tags[];
  public images: Images[];
  public user: User;
  public rating!: Rating[];
  public likeAndDislike: LikeAndDislike[];
  public comments: Comments[];

  constructor() {
    this.overview = new Overview();
    this.tags = new Array<Tags>();
    this.images = new Array<Images>();
    this.user = new User();
    this.likeAndDislike = new Array<LikeAndDislike>();
    this.comments = new Array<Comments>();
  }
}

File: src/main/java/com/recommend/portal/domain/GeneralOverview.java
package com.recommend.portal.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonView;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class GeneralOverview implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @OneToOne
    private Overview overview;
    @ManyToMany
    private List<Tags> tags;
    @OneToMany
    private List<Images> images;
    @OneToOne
    private User user;
    @OneToMany
    private List<Rating> rating;
    @OneToMany
    private List<LikeAndDislike> likeAndDislike;
    @OneToMany
    private List<Comments> comments;

}
